Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -3889,6 +3889,17 @@ JSC_DEFINE_JIT_OPERATION(operationArrayIndexOfValueInt32OrContiguous, UCPUStrict
         OPERATION_RETURN(scope, toUCPUStrictInt32(-1));
     }
 
+    if (searchElement.isInt32()) {
+        for (; index < length; ++index) {
+            JSValue value = data[index].get();
+            if (!value || !value.isInt32())
```

Comment:
What happens if the array has numbers encoded as a double via `jsDoubleNumber`? For example, `jsDoubleNumber(32.0)`, I think, returns false for `JSValue::isInt32` but from JS' perspective, `32 === 32.0`

Reply:
ooops, as you said, `[{}, {}, 3.0, {}].indexOf(3)` results incorrect. I've fixed it. Thank you.